sound/soc/intel/boards/jsl_da7219_max98373.c

```diff
@@ -0,0 +1,421 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright(c) 2019 Intel Corporation.
+
+/*
+ * Intel Jasperlake I2S Machine driver for DA7219 + MAX98373 codec
+ */
+
+#include <linux/dmi.h>
+#include <linux/input.h>
+#include <linux/module.h>
+#include <linux/platform_device.h>
+#include <sound/jack.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/soc-acpi.h>
+#include "../../codecs/da7219.h"
+#include "../../codecs/da7219-aad.h"
+#include "../../codecs/hdac_hdmi.h"
+#include "../common/soc-intel-quirks.h"
+
+#define DIALOG_CODEC_DAI	"da7219-hifi"
+#define MAX98373_CODEC_DAI	"max98373-aif1"
+#define MAXIM_DEV0_NAME		"i2c-MX98373:00"
+#define MAXIM_DEV1_NAME		"i2c-MX98373:01"
+
+#define SOF_DA7219_MCLK_24MHZ	BIT(0)
+
+/* Default: MCLK 24M */
+static unsigned long sof_da7219_quirk = SOF_DA7219_MCLK_24MHZ;
```

Comment:
there is no way this is the case for ICL (38.4)
I would think JSL is the same as GLK (19.2)?

Reply:
The clk was confirmed with Kar before.